### PR TITLE
US25987 load balancer and its configuration

### PR DESCRIPTION
**Pull request by Pushpalatha555**

### CODE REVIEW

File: pkg/client/api.go [hunk 1]
```diff
@@ -50,6 +50,8 @@ func (a *api) do(ctx context.Context, request interface{}, queryParams map[strin
 		panic("api not properly configured")
 	}
 	a.path = fmt.Sprintf("%s/%s/%s", a.client.getHost(), consts.VmaasCmpAPIBasePath, a.path)
+	// a.path = fmt.Sprintf("%s/%s", a.client.getHost(), a.path)
```
Comment (gandharvas) [inline pkg/client/api.go:53]: nit

File: pkg/client/instances_test.go [hunk 2]
```diff
@@ -64,7 +64,7 @@ func TestInstancesAPIService_CloneAnInstance(t *testing.T) {
 					Name:  "Instance_Clone",
 				}
 				// pBody := []byte(`{"zoneId":"1","CloneName":"Instance_Clone"}`)
-				m.EXPECT().getVersion().Return(999999)
+				m.EXPECT().getVersion().Return(999999).AnyTimes()
```
Comment (gandharvas) [inline pkg/client/instances_test.go:67]: why cant it be specific times??

File: pkg/common/constants.go [hunk 3]
```diff
@@ -33,23 +35,28 @@ const (
...
+	LoadBalancerPoolPath           = "pools"
+	LoadBalancerVirtualServersPath = "virtual-servers"
+	NetworkRouterTypePath          = "network-router-types"
+	NetworkServicePath             = "services"
+	NetworkScopePath               = "scopes"
+	ServerPath                     = "servers"
+	VirtualImagePath               = "virtual-images"
+	FolderPath                     = "folders"
+	ConfigOptionPath               = "config-options"
+	PowerSchedulPath               = "power-schedules"
```
Comment (gandharvas) [inline pkg/common/constants.go:52]: typo

File: pkg/common/constants.go [hunk 4]
```diff
@@ -33,23 +35,28 @@ const (
...
+	EnvironmentPath                = "environments"
+	OptionsPath                    = "options"
+	ZoneNetworkOptionsPath         = "zoneNetworkOptions"
+	ProvisionTypesPath             = "provision-types"
+	RouterRoutePath                = "routes"
+	RouterBgpNeighborPath          = "bgp-neighbors"
+	NetworkEdgeClusterPath         = "edge-clusters"
 
 	WhoamiPath = "whoami"
```
Comment (gandharvas) [inline pkg/common/constants.go:61]: to maintain consistency over the convention in the file, please add a comment over whoami path

**Review by gandharvas — COMMENTED** [review]
are these auto-generated via swagger or hand written

**Comment by Pushpalatha555** [comment]
> are these auto-generated via swagger or hand written

hand written

File: pkg/client/api.go (hunk 1 above)
Comment (Pushpalatha555) [reply pkg/client/api.go]: done

File: pkg/client/instances_test.go (hunk 2 above)
Comment (Pushpalatha555) [reply pkg/client/instances_test.go]: done

File: pkg/common/constants.go (hunk 3 above)
Comment (gandharvas) [reply pkg/common/constants.go]: @Pushpalatha555 Could you please change it to 'schedule'.

**Review by gandharvas — APPROVED** [review]
Approving assuming that these changes are validated in the developer's environment.

@Pushpalatha555, please run the ci-suite locally and confirm no issues.

**Review by ravikumar29524 — APPROVED** [review]

**Review by gandharvas — COMMENTED** [review]
LGTM

**Review by gandharvas — APPROVED** [review]

File: pkg/client/load_balancer.go [hunk 5]
```diff
@@ -0,0 +1,410 @@
...
+}
+
+func (lb *LoadBalancerAPIService) CreateLoadBalancer(
+	ctx context.Context,
+	request models.CreateLoadBalancerRequest,
+) (models.CreateNetworkLoadBalancerResp, error) {
+	loadBalancerResp := models.CreateNetworkLoadBalancerResp{}
+	loadBalancerAPI := &api{
+		method: "POST",
+		path:   fmt.Sprintf("%s/%s", consts.BasePath, consts.LoadBalancerPath),
```
Comment (manjunath-batakurki) [inline pkg/client/load_balancer.go:26]: Do we need consts.BasePath ?

Comment (Pushpalatha555) [reply pkg/client/load_balancer.go:26]: yes it will start with api/loadbalancer so BasePath is api

Comment (manjunath-batakurki) [reply pkg/client/load_balancer.go:26]: This is addressed in another review. Resolving the conversation

**Pull request merged**